src/LandIce/evaluators/LandIce_BasalFrictionCoefficient_Def.hpp

```diff
@@ -228,6 +235,258 @@ postRegistrationSetup (typename Traits::SetupData d,
     memoizer.enable_memoizer();
 }
 
+// *********************************************************************
+// Kokkos functor (Cell)
+template<typename EvalT, typename Traits, typename EffPressureST, typename VelocityST, typename TemperatureST>
+KOKKOS_INLINE_FUNCTION
+void BasalFrictionCoefficient<EvalT, Traits, EffPressureST, VelocityST, TemperatureST>::
+operator() (const GivenFieldParam_Tag& tag, const int& cell) const {
+  for (unsigned int ipt=0; ipt<dim; ++ipt)
+    beta(cell,ipt) = given_field_param(cell,ipt);
+}
+template<typename EvalT, typename Traits, typename EffPressureST, typename VelocityST, typename TemperatureST>
+KOKKOS_INLINE_FUNCTION
+void BasalFrictionCoefficient<EvalT, Traits, EffPressureST, VelocityST, TemperatureST>::
+operator() (const GivenField_Tag& tag, const int& cell) const {
+  for (unsigned int ipt=0; ipt<dim; ++ipt)
+    beta(cell,ipt) = given_field(cell,ipt);
```

Comment:
I'm not sure. The scalar types of the two fields are different, so deep_copy might either not compile, or do a memcpy (which would be even worse). I guess we could try and though.

Reply:
Good point, I did not see the different scalar type. I'm not sure if sacado has the overloads for this case. I would hope there's something in place to prevent a direct memcpy but maybe there's no such guard.

Reply:
It looks like we will have to mess with this a bit more to get it to compile so I'm marking resolved for now and we can revisit in the future.